lib/ash/resource/change/cascade_destroy.ex

```diff
@@ -0,0 +1,204 @@
+defmodule Ash.Resource.Change.CascadeDestroy do
+  @option_schema [
+    relationship: [
+      type: :atom,
+      doc: "The name of the relationship to work on",
+      required: true
+    ],
+    resource: [
+      type: {:spark, Ash.Resource},
+      doc: "The resource to introspect for relationship information",
+      required: true
+    ],
+    action: [
+      type: :atom,
+      doc: "The name of the destroy action to call on the related resource",
+      required: false,
+      default: :destroy
+    ]
+  ]
+
+  @moduledoc """
+  Cascade a resource's destroy action to a related resource's destroy action.
+
+  Adds an after-action hook that explicitly calls destroy on any records related
+  via the named relationship.  It will optimise for bulk destroys where
+  possible.
+
+  > #### Beware database constraints {: .warning}
+  >
+  > Think carefully before using this change with data layers which enforce
+  > referential integrity (ie PostgreSQL and SQLite) and you may need to defer
+  > constraints for the relationship in question.
+  >
+  > See also:
+  >   1. [PostgreSQL's `SET CONSTRAINTS` documentation](https://www.postgresql.org/docs/current/sql-set-constraints.html)
+  >   2. [SQLite's `PRAGMA defer_foreign_keys` documentation](https://www.sqlite.org/pragma.html#pragma_defer_foreign_keys)
+
+  ## Options
+
+  #{Spark.Options.docs(@option_schema)}
+
+  ## Example
+
+      change {Ash.Resource.Change.CascadeDestroy, resource: __MODULE__, relationship: :comments, action: :destroy}
```

Comment:
do you actually need the `resource` constraint to be set in the resource? Is this something we do for you? If not, you should always have a way to get the resource in the callbacks surely.

Reply:
yeah, but we can't validate the options in the `init` callback if we don't have it.

Reply:
We should probably do something in the framework to provide that constraint or something then TBH. Don't want people to have to pass the current resource in. Also, unfortunately, at the moment the init callback is not actually called to initialize and then persisted or anything. The init is called every time that change is called. I intend to improve that in the future.